### DISCUSSION TITLE
Question about AppStream and Software Center integration

### DISCUSSION THREAD

**Discussion by kuiperzone** [AppImage/AppImageKit#1252] [Q&A]
Hi,

I'm developing a deployment utility. I've got it working to create an AppImage and this isn't strictly an AppImage question -- but there are few places I can ask it and, I suspect if anyone can answer, I will find those people here. I have to say that I am at my wits end with this, as I cannot diagnose the problem.

I've extended the utility to create an RPM with AppStream data. When I install the RPM (with a test "hello world" app), it all (almost) works fine and appears in the Gnome Software Center.

While it can be installed and removed with `dnf install/remove`, it cannot be launched or removed from the Software Center, as the buttons do nothing. See screenshot.

I was of the understanding that it put my desktop and metainfo.xml in the right places under the right names, the Software Center will pick these up.

I'm wondering, if I paste my desktop and metainfo.xml files below, could someone tell me what I am missing?

Thanks

![Screenshot 1](https://user-images.githubusercontent.com/17156238/222987244-a6a65979-b3be-4606-a7b2-a2cf35c50396.png)

**/usr/share/applications/net.example.hellopuppy.desktop**

    [Desktop Entry]
    Type=Application
    Name=Hello Puppy
    Icon=net.example.hellopuppy
    Comment=A HelloWorld application
    Exec=/opt/net.example.hellopuppy/HelloPuppy
    TryExec=/opt/net.example.hellopuppy/HelloPuppy
    Terminal=true
    Categories=Utility
    MimeType=
    Keywords=

**/usr/share/metainfo/net.example.hellopuppy.metainfo.xml**

    <?xml version="1.0" encoding="UTF-8"?>
    <component type="desktop">
    
        <id>net.example.hellopuppy</id>
        <name>HelloPuppy</name>
        <summary>A HelloWorld application</summary>
        <developer_name>Acme Ltd</developer_name>
        <url type="homepage">https://example.net</url>
        <metadata_license>MIT</metadata_license>
        <project_license>GPL-3.0-or-later</project_license>
    
        <content_rating type="oars-1.1" />
    
        <description>
            <p>This is a longer application description. REPLACE THIS WITH YOUR OWN.
            IMPORTANT: In this file, you can use supported macros. See the --help output for details.</p>
        </description>
    
        <releases>
            <release version="1.0.0" date="2023-03-04">
                <description><p>The latest release.</p></description>
            </release>
        </releases>
    
        <screenshots>
            <screenshot type="default">
                <image>https://i.postimg.cc/0jc8xxxC/Hello-Computer.png</image>
            </screenshot>
        </screenshots>
    
        <launchable type="desktop-id">net.example.hellopuppy.desktop</launchable>
    
    </component>

**Comment by probonopd** [comment]
AppImages are relocatable in the filesystem, which means the user can place them in any location. So, how would an "Open" button or a "Delete" button in a GUI know where the user has put the AppImage file?

I keep mine on USB sticks, optical media (for archival), and on network shares. And in many versions and variants. This is why app stores and software centers never work for me, as they always tend to assume that one has only one version of each appliation (usually: the latest), at one fixed location. This assumption is anything but true for AppImages.

**Reply by kuiperzone** [reply]
Hi, Thanks. As I said, it's not an AppImage, but an RPM. I know that this is an AppImage forum, but I honestly don't know where to ask this and I suspect my problem might a AppStream data issue. I'm writing a deployment utility -- I got AppImage working.

**Reply by probonopd** [reply]
Oh sorry; I didn't catch that. I don't know.

**Reply by kuiperzone** [reply]
No worries. Glad of the reply.

**Comment by kuiperzone** [comment]
To add, if anyone has any ideas of where else I could this question (if not here), it would be appreciated also.

**Reply by probonopd** [reply]
https://gitlab.gnome.org/GNOME/gnome-software/-/issues possibly?
